src/lib.rs

```diff
@@ -271,96 +280,165 @@ fn filter_attr(attr: &&syn::Attribute) -> bool {
 
 fn builder_for_struct(ast: syn::MacroInput) -> quote::Tokens {
     debug!("Deriving Builder for '{}'.", ast.ident);
-    let opts = Options::from(&ast.attrs);
-    if !opts.setter_enabled() {
-        trace!("Setters disabled for '{}'.", ast.ident);
-        return quote!();
-    }
+    let opts = StructOptions::from(&ast);
 
     let fields = match ast.body {
         syn::Body::Struct(syn::VariantData::Struct(ref fields)) => fields,
         _ => panic!("#[derive(Builder)] can only be used with braced structs"),
     };
 
-    let name = &ast.ident;
+    let struct_name = &ast.ident;
+    let builder_name = syn::Ident::from(opts.builder_name());
     let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
-    let funcs = fields.iter().map(|f| {
 
-        trace!("Parsing field {:?}.", f.ident.as_ref().map(|i| i.as_ref()).unwrap_or_default());
+    let mut setter_fns     = Vec::<quote::Tokens>::with_capacity(fields.len());
+    let mut builder_fields = Vec::<quote::Tokens>::with_capacity(fields.len());
+    let mut initializers   = Vec::<quote::Tokens>::with_capacity(fields.len());
+
+    for f in fields {
+        let name = f.ident.as_ref()
+            .expect(&format!("Missing identifier for field {:?}.", f))
+            .as_ref();
+        trace!("Parsing field {}.", name);
 
         let f_opts = OptionsBuilder::<FieldMode>::default()
             .parse_attributes(&f.attrs)
-            .with_struct_options(&opts);
-
-        debug!("Setter prefix is {:?}", f_opts.setter_prefix());
+            .build(name, &opts);
 
-        let mut tokens = quote::Tokens::new();
+        trace!("Filtering field attributes");
+        let attrs: Vec<&syn::Attribute> = f.attrs.iter()
+            .filter(|a| {
+                let keep = filter_attr(a);
+                match keep {
+                    true => trace!("Keeping field attribute for builder field and setter {:?}", a),
+                    false => trace!("Ignoring field attribute for builder field and setter {:?}", a)
+                }
+                keep
+            }).collect();
 
-        if f_opts.setter_enabled() {
-            derive_setter(f, &f_opts).to_tokens(&mut tokens);
-        } else {
-            trace!("Skipping setter.");
-        }
+        setter_fns.push(derive_setter(f, &f_opts, &attrs));
+        builder_fields.push(derive_builder_field(f, &f_opts, &attrs));
+        initializers.push(derive_initializer(f, &f_opts));
+        // NOTE: Looking forward for computation in interpolation
+        // - https://github.com/dtolnay/quote/issues/10
```

Comment:
Linking here to trigger Githubs magic cross linkery: https://github.com/dtolnay/quote/issues/10